pyecoregen/ecore.py

```diff
@@ -269,6 +297,44 @@ def create_environment(self, **kwargs):
 
         return environment
 
+    @staticmethod
+    def resolve_all_proxies(root: ecore.EObject):
+        """Resolves all the proxies from an object"""
+        for o in root.eAllContents():
+            if isinstance(o, ecore.EEnum):
+                continue
+            for reference in o.eClass.eAllReferences():
+                values = o.eGet(reference)
+                if not reference.many:
+                    values = [values]
+                for value in values:
+                    if not isinstance(value, ecore.EProxy):
+                        continue
+                    value.eResource  # force proxy resolution
+
+    @staticmethod
+    def load_all_required_resources(root):
+        """
+        Returns a set of all the resources on which depends a model root.
+
+        Args:
+            root: the model root that will be traversed to find dependencies.
+        """
+        rset = root.eResource.resource_set
+        before = {v for _, v in rset.resources.items()}
+        for x in before:
+            EcoreGenerator.resolve_all_proxies(x.contents[0])
+        current = {v for _, v in rset.resources.items()}
```

Comment:
Could you use `values()` here instead of `items()`?

Reply:
Done!